pht_federated/aggregator/api/endpoints/discovery.py

```diff
@@ -0,0 +1,117 @@
+            discovery_item_count += discovery['item_count']
+            discovery_feature_count += discovery['feature_count']
+
+            for feature in discovery['column_information']:
+                if query:
+                    selected_features = query.split(',')
+                    if feature['title'] in selected_features:
+                        feature_lst.append(feature)
+                else:
+                    feature_lst.append(feature)
+
+        for feature in feature_lst:
+            if feature['type'] == 'numeric':
+                discovery_summary_json = aggregate_numeric_column(feature_lst, feature)
+                aggregated_feature_lst.append(discovery_summary_json)
+
+            elif feature['type'] == 'categorical':
+                discovery_summary_json = aggregate_categorical_column(feature_lst, feature, len(response))
+                aggregated_feature_lst.append(discovery_summary_json)
+
+            elif feature['type'] == 'unstructured':
+                discovery_summary_json = aggregate_unstructured_data(feature_lst, feature)
+                aggregated_feature_lst.append(discovery_summary_json)
+
+            elif feature['type'] == 'unique':
+                discovery_summary_json = aggregate_unique_column(feature_lst, feature, len(response))
+                aggregated_feature_lst.append(discovery_summary_json)
+
+            elif feature['type'] == 'equal':
+                discovery_summary_json = aggregate_equal_column(feature_lst, feature)
+                aggregated_feature_lst.append(discovery_summary_json)
+
+            feature_lst = [x for x in feature_lst if x['title'] != feature['title']]
+
+            if len(feature_lst) == 0:
+                break
+
+        discovery_feature_count /= len(response)
+
+        discovery_summary_schema = {
+            "proposal_id": proposal_id,
+            "item_count": discovery_item_count,
+            "feature_count": discovery_feature_count,
+            "column_information": aggregated_feature_lst
+        }
+        discovery_summary = DiscoverySummary(**discovery_summary_schema)
+
+        print("DISCOVERY SUMMARY : {}".format(discovery_summary))
+
+        return discovery_summary
+
+
+@router.delete("/{proposal_id}/discovery", response_model=DiscoveryStatistics)
+def delete_discovery_statistics(proposal_id: int, db: Session = Depends(dependencies.get_db)) -> DiscoveryStatistics:
+    discovery_del = datasets.delete_by_proposal_id(proposal_id, db)
+    if not discovery_del:
+        raise HTTPException(status_code=404, detail=f"DatasetStatistics of proposal with id '{proposal_id}' not found.")
+    return discovery_del
+
+@router.post("/{proposal_id}/discovery", response_model=DiscoveryStatistics)
+def post_discovery_statistics(proposal_id: int, create_msg: StatisticsCreate, db: Session = Depends(dependencies.get_db)) -> DatasetStatistics:
+    dataset_statistics = json.loads(create_msg.json())
```

Comment:
avoid unnecessary conversion. Use class methods of pydantic models in this case `create_msg.dict()` or use class fields directly. `json` module should not be used with pydantic models